x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/StringUtilsTests.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.ql.expression.predicate.regex.LikePattern;
+
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.elasticsearch.xpack.eql.util.StringUtils.toLikePattern;
+
+public class StringUtilsTests extends ESTestCase {
+
+    public void testLikePatternNoPattern() throws Exception {
+        String string = "abc123";
+        assertEquals(string, toLikePattern(string).asString());
+    }
+
+    public void testLikePatternLikeChars() throws Exception {
+        String string = "a%bc%%12_3__";
+        String escape = Character.toString(1);
```

Comment:
Not strictly related to the change, but was wondering the value choice for the escape character (in `StringUtils#toLikePattern()`). Would `\0` have collided with anything?

Reply:
0 ignores escaping, that is no character acts as an escape.